src/libs/actions/Report.js

```diff
@@ -904,7 +904,9 @@ function editReportComment(reportID, originalReportAction, textForNewComment) {
 
     // Do not autolink if someone explicitly tries to remove a link from message.
     // https://github.com/Expensify/App/issues/9090
-    const htmlForNewComment = parser.replace(textForNewComment, {filterRules: _.filter(_.pluck(parser.rules, 'name'), name => name !== 'autolink')});
+    const autolinkFilter = {filterRules: _.filter(_.pluck(parser.rules, 'name'), name => name !== 'autolink')};
+    const htmlForNewComment = parser.replace(textForNewComment, autolinkFilter);
+    const originalMessageHTML = parser.replace(originalReportAction.message[0].html, autolinkFilter);
```

Comment:
@stitesExpensify coming from https://github.com/Expensify/App/issues/13661, I don't believe this PR caused this original issue (because there's some issue with escaping spaces as well) but I am seeing that by passing HTML into `parser.replace` we're effectively treating the HTML like markdown, is that not correct?
<img width="1555" alt="Screenshot 2022-12-19 at 12 01 44 PM" src="https://user-images.githubusercontent.com/4741899/208510292-f84f9065-18af-435e-9635-ec4eb581786b.png">

Reply:
Okay yes I can confirm, PR incoming

Reply:
I think this should resolve it: https://github.com/Expensify/App/pull/13703

Reply:
> we're effectively treating the HTML like markdown, is that not correct?

I don't really understand the question TBH.

In this issue, the problem (at least I thought) was that we were comparing parsed HTML to unparsed HTML, so when the strings were the same it was failing because it was comparing `"` to `&quot;`, which is why I parsed the original HTML so that we were comparing the same things. 

Looking at your screenshot, there are 3 different interpretations of the same HTML which seems super messy 😅 I think ideally we would want to just compare the `originalCommentHTML` to the `htmlForNewComment` but even _those_ are in different formats.